### PR TITLE
feat(maillog): Add errornous record filter for mail log records

### PR DESCRIPTION
**Pull request by patrickvenetz**
This Pull Request adds `filters: { errornous: TRUE|FALSE }` to mail log queries.

```gql
{
  mailLog(filters: {errornous: true}) {
    totalCount
    nodes {
      email
      error
    }
  }
  me {
    mailLog(filters: {errornous: false}) {
      totalCount
      nodes {
        error
      }
    }
  }
}
```

Code is slightly refactored and extended, moving pagination (including filtering) into a lib-file as well as record props conditionals (error, status, template).

### CODE REVIEW

File: packages/maillog/graphql/schema-types.js [hunk 1]
```diff
@@ -1,5 +1,9 @@
 module.exports = `
 
+input MaiLogFiltersInput {
+  errornous: Boolean
```
Comment (patte) [inline packages/maillog/graphql/schema-types.js:4]: typo, the thing is spelled: `erroneous`. Why not just `error`?

File: packages/maillog/lib/paginate.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  return stack
+}
+
+const paginate = (records, args) => paginator(
+  { first: 100, ...args },
+  a => a,
+  ({ filters = {} }) => {
+    const filterFns = getFilterFns(filters)
+
+    return records.filter(record => filterFns.every(filterFn => filterFn(record)))
```
Comment (patte) [inline packages/maillog/lib/paginate.js:23]:
```suggestion
    return filterFns.length
      ? records.filter(record => filterFns.every(filterFn => filterFn(record)))
      : records
```

File: packages/maillog/graphql/schema-types.js (hunk 1 above)
Comment (patrickvenetz) [reply packages/maillog/graphql/schema-types.js]: `error` won't indicate a `Boolean` value, considering `MailLogRecord.error` is a `String`.

Settle for `hasError`!

File: packages/maillog/lib/paginate.js (hunk 2 above)
Comment (patrickvenetz) [reply packages/maillog/lib/paginate.js]: Thanks!

**Comment by patrickvenetz** [comment]
@patte Please revisit.

**Pull request merged**